From 93520bd488c49644016e0f16a80c1c954fef7fe5 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 14 Sep 2020 09:05:46 +0300
Subject: [PATCH] [SW-17444] preboot: fit: remove usage of public key property

Currently, in order to get public key node as part of secure boot flow,
preboot tries to get pub-key name as property and then retrieves
public key node from fit.

Since public key name is pre-defined, this patch add changes to pass
public key string directly and retrieve public key node thus avoiding
adding public key as property in .its file

Change-Id: I4e6f31dcf898de9d9f7e2c1ee94f448f646bba10
---

diff --git a/preboot/fit.c b/preboot/fit.c
index afffd9d..4782c4a 100644
--- a/preboot/fit.c
+++ b/preboot/fit.c
@@ -69,13 +69,6 @@
 	return fdt_stringlist_get(fit, noffset, FIT_LOADABLE_PROP, id, NULL);
 }
 
-static const char *fit_pub_key_name(const void *fit, int noffset)
-{
-	int prop_len;
-
-	return fdt_getprop(fit, noffset, FIT_PUB_KEY_PROP, &prop_len);
-}
-
 static int fit_find_config_node(const void *fit)
 {
 	int conf, node;
@@ -290,27 +283,17 @@
 static int fit_pub_key_node_get(const void *fit, int conf)
 {
 	int images;
-	const char *uname = fit_pub_key_name(fit, conf);
 
 	images = fdt_path_offset(fit, FIT_IMAGES_PATH);
-	if (uname)
-		return fdt_subnode_offset(fit, images, uname);
-
-	return -1;
+	return fdt_subnode_offset(fit, images, FIT_PUB_KEY_PROP);
 }
 
 static int fit_pub_key_ext_node_get(const void *fit, int conf)
 {
-	int images, prop_len;
-	const char *uname;
-
-	uname = fdt_getprop(fit, conf, FIT_PUB_KEY_EXT_PROP, &prop_len);
+	int images;
 
 	images = fdt_path_offset(fit, FIT_IMAGES_PATH);
-	if (uname)
-		return fdt_subnode_offset(fit, images, uname);
-
-	return -1;
+	return fdt_subnode_offset(fit, images, FIT_PUB_KEY_EXT_PROP);
 }
 
 static int fit_get_pub_key(const void *fit, int noffset, int sig_sub_node,
